src/python_testing/TC_ACE_1_4.py

```diff
@@ -0,0 +1,344 @@
+        asserts.assert_true(cluster in results[endpoint], err_msg)
+        err_msg = "Results not returned for attribute {} on ep {}".format(str(attribute), str(endpoint))
+        asserts.assert_true(attribute in results[endpoint][cluster], err_msg)
+        attr_ret = results[endpoint][cluster][attribute]
+        err_msg = "Error reading {}:{}".format(str(cluster), str(attribute))
+        asserts.assert_true(attr_ret is not None, err_msg)
+        asserts.assert_false(isinstance(attr_ret, Clusters.Attribute.ValueDecodeFailure), err_msg)
+
+    @async_test_body
+    async def test_TC_ACE_1_3(self):
+        # TODO: Guard these on the PICS
+        asserts.assert_true('PIXIT.ACE.APPENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT.ACE.APPENDPOINT must be included on the command line in the --int-arg flag as PIXIT.ACE.APPENDPOINT:<endpoint>")
+        asserts.assert_true('PIXIT.ACE.APPCLUSTER' in self.matter_test_config.global_test_params,
+                            "PIXIT.ACE.APPCLUSTER must be included on the command line in the --string-arg flag as PIXIT.ACE.APPCLUSTER:<cluster_name>")
+        asserts.assert_true('PIXIT.ACE.APPATTRIBUTE' in self.matter_test_config.global_test_params,
+                            "PIXIT.ACE.APPATTRIBUTE must be included on the command line in the --string-arg flag as PIXIT.ACE.APPATTRIBUTE:<attribute_name>")
+        asserts.assert_true('PIXIT.ACE.APPDEVTYPEID' in self.matter_test_config.global_test_params,
+                            "PIXIT.ACE.APPDEVTYPEID must be included on the command line in the --int-arg flag as PIXIT.ACE.APPDEVTYPEID:<device_type_id>")
+
+        cluster_str = self.matter_test_config.global_test_params['PIXIT.ACE.APPCLUSTER']
+        attribute_str = self.matter_test_config.global_test_params['PIXIT.ACE.APPATTRIBUTE']
+
+        self.cluster = str_to_cluster(cluster_str)
+        self.attribute = str_to_attribute(self.cluster, attribute_str)
+        self.devtype = self.matter_test_config.global_test_params['PIXIT.ACE.APPDEVTYPEID']
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT.ACE.APPENDPOINT']
+
+        asserts.assert_true(self.cluster is not None, "Invalid cluster name")
+        asserts.assert_true(self.attribute is not None, "Invalue attribute name")
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "TH1 writes ACL all clusters view on all endpoints")
+        admin_acl = Clusters.AccessControl.Structs.AccessControlEntry(privilege=Clusters.AccessControl.Enums.Privilege.kAdminister,
+                                                                      authMode=Clusters.AccessControl.Enums.AuthMode.kCase,
+                                                                      subjects=[],
+                                                                      targets=[Clusters.AccessControl.Structs.Target(endpoint=0, cluster=0x001f)])
```

Comment:
nit, can this magic cluster numbers be named constants? (here and all other spots of `cluster=` and `deviceType=`

Reply:
I changed the ones for the clusters, but I'm not sure how one would access a list of device type IDs in python. Happy to make the change if you can point me at the location.

Note that we do have these hardcoded in the test case, so there is a reference for these, even if it's not in the script.

Reply:
Spoke with Terence offline and the device type enums would need more code gen to create. Leaving this for now as this is an orthogonal problem.